lib/stdlib/src/epp.erl

```diff
@@ -345,17 +347,42 @@ parse_file(Ifile, Options) ->
       WarningInfo :: warning_info().
 
 parse_file(Epp) ->
-    case parse_erl_form(Epp) of
-	{ok,Form} ->
-            [Form|parse_file(Epp)];
+    %% Code duplicated from parse_erl_form(Epp), but with
+    %% added search for tokens to warn for
+    case epp_request(Epp, scan_erl_form) of
+	{ok,Toks} ->
+            Warnings = parse_file_warnings(Toks),
+            case erl_parse:parse_form(Toks) of
+                {ok, Form} ->
+                    [Form|Warnings] ++ parse_file(Epp);
+                Problem2 ->
+                    parse_file_problem(Epp, Problem2, Warnings)
+            end;
+        Problem1 ->
+            parse_file_problem(Epp, Problem1, [])
+    end.
+
+parse_file_problem(Epp, Problem, Warnings) ->
+    case Problem of
 	{error,E} ->
-	    [{error,E}|parse_file(Epp)];
+	    [{error,E}|Warnings] ++ parse_file(Epp);
 	{warning,W} ->
-	    [{warning,W}|parse_file(Epp)];
+	    [{warning,W}|Warnings] ++ parse_file(Epp);
 	{eof,Location} ->
-	    [{eof,Location}]
+	    [{eof,Location}|Warnings]
     end.
 
+parse_file_warnings(Toks) ->
```

Comment:
Would it make more sense to re-write this function in fold-style /filtermap?
The only reason why I mention this is that a recursive function can do many things, but a filtermap / foldl is what is, and it is more strict in its semantics.

```erlang
parse_file_warnings(Toks) ->
  lists:filtermap(fun ({tqstring,Anno,_}) -> 
                         Winfo = {erl_anno:location(Anno),?MODULE,tqstring}},
                         {true, {warning, Winfo};
                      (_) -> false
                  end, Toks).
```

Reply:
I have never gotten accustomed to `lists:filtermap/2`, but sure, it is a valid point.
I will probably do it at the beginning of next week...

Reply:
A list comprehension would also work, and would be more concise than a filtermap.

Reply:
TBH, this is a suggestion, non mandatory. The changes are great!